backend/elasticsearch-aws/src/main/java/org/hibernate/search/backend/elasticsearch/aws/cfg/ElasticsearchAwsBackendSettings.java

```diff
@@ -6,12 +6,15 @@
  */
 package org.hibernate.search.backend.elasticsearch.aws.cfg;
 
+import org.hibernate.search.configuration.properties.collector.HibernateSearchConfiguration;
+
 /**
  * AWS-specific configuration properties for Elasticsearch backends.
  * <p>
  * Constants in this class are to be appended to a prefix to form a property key;
  * see {@link org.hibernate.search.engine.cfg.BackendSettings} for details.
  */
+@HibernateSearchConfiguration(prefix = "hibernate.search.backends.")
```

Comment:
My parsing logic fails to find a difference between the first two bullets (both point to `*BackendSettings`) 😄 🙈

Reply:
Looks like my writing logic failed; I updated my comment :)

Reply:
But maybe you intended to extract this processor and use it in other places (e.g. Hibernate ORM)?
In that case it makes sense to rely on explicit attributes rather than a hardcoded logic.

It would also make sense to keep the explicit `type` attribute, since not every project follows the same package naming conventions and `AUTO` might not work everywhere. Also some packages in ORM are misnamed and I suspect some SPI properties are mixed with API properties, so they would need overrides.

Reply:
I didn't want to super tie it to Search, so that's why there are those config properties or the annotation. Otherwise, yeah.. we could've put most of the filtering hardcoded into the processor itself 😃.
I've removed the type from the annotation, but the enum is still there. 
As for the ORM - if it could be of use there, I can try to see how we could adapt this processor to work with ORM too.
And thanks for updating the comment 👍🏻🏆

Reply:
>  As for the ORM - if it could be of use there, I can try to see how we could adapt this processor to work with ORM too.

Better check with Steve if he finds that useful. But yeah IMO that could be nice.

Reply:
let's get it working nicely here and then show what we've got 😃